Source/IsomorphicMassAssign.cpp

```diff
@@ -483,6 +501,19 @@ void IsomorphicMassAssign::mappingLogicChanged(MappingLogicBase* mappingLogicTha
 	}
 }
 
+void IsomorphicMassAssign::scaleStructurePeriodChanged(int newPeriod)
+{
+	scaleSizeBox->setSelectedId(newPeriod, dontSendNotification);

```

Comment:
You might also call the function with sendNotification, letting the latter do the necessary updates. (Unless this creates an endless loop...)

Reply:
You're right! When I first made the PR it would have created a loop, but the improvements I made after prevent that. I'll use sendNotifcation for now.